src/cleanlab_tlm/utils/rag.py

```diff
@@ -135,6 +140,45 @@ def __init__(
 
         _validate_trustworthy_rag_options(options=options, initialized_evals=self._evals)
 
+        # Optional per-eval tool call overrides
+        # These are name-based include/exclude sets used only in the _handle_tool_call_filtering decorator
+        self._configure_tool_call_eval_overrides(exclude_names=[k.name for k in self._evals if k.response_identifier])
+
+    def _configure_tool_call_eval_overrides(
+        self,
+        *,
+        exclude_names: Optional[list[str]] = None,
+    ) -> None:
+        """Configure per-eval overrides for tool call handling.
+
+        This method allows fine-grained control over which evals are processed vs.
+        skipped (scored as None) when the response is detected as a tool call, without
+        modifying the Eval objects.
+
+        - If an eval name is in exclude_names, it will be filtered (score=None) during tool call handling.
+
+        Args:
+            exclude_names (list[str] | None): Evaluation names to always filter during tool calls.
+                This replaces the existing exclude set.
+        """
+        exclude_names = exclude_names or []
+
+        def _is_valid_name(name: str) -> bool:
+            is_eval_name = any(eval_obj.name == name for eval_obj in self._evals)
+            eval_obj = next((eval_obj for eval_obj in self._evals if eval_obj.name == name), None)
+            has_response_identifier = eval_obj is not None and eval_obj.response_identifier is not None
```

Comment:
Can you make it clearer in the docstring why we are also checking for the `response_identifier` key here? (without context it is not immediately clear why this key has to exist for it to be valid)